internal/build/job/manager.go

```diff
@@ -47,19 +54,75 @@ func (jbm *jobManager) GarbageCollect(ctx context.Context, mod kmmv1beta1.Module
 	return deleteNames, nil
 }
 
-func (jbm *jobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, targetImage string, pushImage bool) (build.Result, error) {
+func (jbm *jobManager) ShouldSync(
+	ctx context.Context,
+	mod kmmv1beta1.Module,
+	m kmmv1beta1.KernelMapping,
+	opts ...build.SyncOption) (bool, error) {
```

Comment:
why not just pass a namespace parameter to this function? We are construction all those options, but all we care about is just a namespace. The code that is calling this function should now which namespace to use ( it is probably the namespace of the Module)